src/plugins/console/server/routes/api/console/proxy/create_handler.ts

```diff
@@ -75,7 +75,11 @@ function toUrlPath(path: string) {
   // in JS can lead to data loss (7.0 will get munged into 7, thus losing indication of
   // measurement precision)
   if (!urlPath.includes('?pretty')) {
-    urlPath += '?pretty=true';
+    if (urlPath.includes('?')) {
+      urlPath += '&pretty=true';
+    } else {
+      urlPath += '?pretty=true';
+    }
```

Comment:
What do you think of rewriting this `toUrlPath()` function to the below code? I think this will help us avoid most corner cases failure.
```
function toUrlPath(path: string) {
  const FAKE_BASE = 'http://localhost';
  const urlWithFakeBase = new URL(`${FAKE_BASE}/${trimStart(path, '/')}`);
  // Appending pretty here to have OpenSearch do the JSON formatting, as doing
  // in JS can lead to data loss (7.0 will get munged into 7, thus losing indication of
  // measurement precision)
  if (!urlWithFakeBase.searchParams.get('pretty')) {
    urlWithFakeBase.searchParams.append('pretty', 'true');
  }
  const urlPath = urlWithFakeBase.href.replace(urlWithFakeBase.origin, '');
  return urlPath;
}
```

Reply:
this looks much better and clear.